From 24d9e9c66cb41b303f62ca803cbaf17ebf5eb79e Mon Sep 17 00:00:00 2001
From: "H. Peter Anvin" <hpa@linux.intel.com>
Date: Tue, 21 Feb 2012 17:25:50 -0800
Subject: [PATCH 2/2] x86, cpufeature: Add CPU features from Intel document
 319433-012A
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
Patch-mainline: v3.4
Git-commit: 513c4ec6e4759aa33c90af0658b82eb4d2027871
References: fate#313826

Add CPU features from the Intel Archicture Instruction Set Extensions
Programming Reference version 012A (Feb 2012), document number 319433-012A.

Signed-off-by: H. Peter Anvin <hpa@linux.intel.com>
(cherry picked from commit 513c4ec6e4759aa33c90af0658b82eb4d2027871)

[AF: Backported (no X86_FEATURE_BMI1 or X86_FEATURE_AVX2)]
Signed-off-by: Andreas Färber <afaerber@suse.de>
---
 arch/x86/include/asm/cpufeature.h |    3 +++
 1 Datei geändert, 3 Zeilen hinzugefügt(+)

diff --git a/arch/x86/include/asm/cpufeature.h b/arch/x86/include/asm/cpufeature.h
index 06ac811..c728609 100644
--- a/arch/x86/include/asm/cpufeature.h
+++ b/arch/x86/include/asm/cpufeature.h
@@ -197,8 +197,11 @@
 
 /* Intel-defined CPU features, CPUID level 0x00000007:0 (ebx), word 9 */
 #define X86_FEATURE_FSGSBASE	(9*32+ 0) /* {RD/WR}{FS/GS}BASE instructions*/
+#define X86_FEATURE_HLE		(9*32+ 4) /* Hardware Lock Elision */
 #define X86_FEATURE_SMEP	(9*32+ 7) /* Supervisor Mode Execution Protection */
 #define X86_FEATURE_ERMS	(9*32+ 9) /* Enhanced REP MOVSB/STOSB */
+#define X86_FEATURE_INVPCID	(9*32+10) /* Invalidate Processor Context ID */
+#define X86_FEATURE_RTM		(9*32+11) /* Restricted Transactional Memory */
 
 #if defined(__KERNEL__) && !defined(__ASSEMBLY__)
 
-- 
1.7.10.4

